tool/randomizer.lua

```diff
@@ -32,27 +32,49 @@ for proficiency, deps in pairs(dependencies) do
     end
 end
 
+randomizer.seedqueue = {}
 randomizer.seed = ""
+randomizer.seedtype = ""
 randomizer.proficiency = "beginner"
 randomizer.proficiencies = { "beginner", "intermediate", "advanced" }
+randomizer.newgamenewseed = "Auto"
+randomizer.newgamenewseedvalues = { "On", "Off", "Auto" }
+randomizer.newgamenewseedui = { ["On"] = "ON", ["Off"] = "OFF", ["Auto"] = "Auto (ON for Unseeded / OFF for Seeded)" }
 
 local levelsequence
-local levelindex
+local levelindex = 0
 
-function randomizer.drawhud()
-    ui.drawlines({ "Randomizer " .. randomizer.proficiency .. " Seed " .. tostring(randomizer.seed) })
+function randomizer.hudlines()
+    local randomizerlines = {}
+    local firstline = "Randomizer " .. randomizer.proficiency .. " " .. randomizer.seedtype
+    if randomizer.seed ~= "" then
+        firstline = firstline .. ": " .. randomizer.seed
+    end
+    table.insert( randomizerlines, firstline )
+    if levelindex == 0 then
+        table.insert( randomizerlines, "Press New Game to start" )
+    elseif levelindex > 1 then
+        table.insert( randomizerlines, "Progress " .. levelindex - 2 .. "/" .. #levelsequence + 1 )
+    end
```

Comment:
What happens on `levelindex == 1`? Also when the game is finished, maybe a different message should be displayed?

Reply:
levelindex can now only have a value of 1 or greater. `levelindex == 1` is its value before the randomizer starts, but whenever this happens the progress will not be displayed and "Press New Game to start" will be displayed instead.